Add `regexp/no-empty-character-class` rule (#329)
ota-meshi/eslint-plugin-regexp: docs/rules/no-empty-character-class.md

---
pageClass: "rule-details"
sidebarDepth: 0
title: "regexp/no-empty-character-class"
description: "disallow character classes that does not match all characters"
---
# regexp/no-empty-character-class

> disallow character classes that does not match all characters

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>

## :book: Rule Details

This rule reports character classes that cannot match all characters.

> The character class that cannot match all characters is an empty character class and a negative character class that contains all characters.

Reviewer: ```suggestion
Character classes that cannot match any characters are either empty or negated character classes of elements that contain all characters.
```

pageClass: "rule-details"
sidebarDepth: 0
title: "regexp/no-empty-character-class"
description: "disallow character classes that does not match all characters"
---
# regexp/no-empty-character-class

> disallow character classes that does not match all characters

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>

## :book: Rule Details

> This rule reports character classes that cannot match all characters.

Reviewer: ```suggestion
This rule reports character classes that cannot match any characters.
```
